components/scream/src/physics/p3/tests/p3_rain_imm_freezing_unit_tests.cpp

```diff
@@ -0,0 +1,136 @@
+#include "catch2/catch.hpp"
+
+#include "share/scream_types.hpp"
+#include "share/util/scream_utils.hpp"
+#include "share/scream_kokkos.hpp"
+#include "share/scream_pack.hpp"
+#include "physics/p3/p3_functions.hpp"
+#include "physics/p3/p3_functions_f90.hpp"
+#include "share/util/scream_kokkos_utils.hpp"
+
+#include "p3_unit_tests_common.hpp"
+
+#include <thread>
+#include <array>
+#include <algorithm>
+#include <random>
+
+namespace scream {
+namespace p3 {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestRainImmersionFreezing {
+
+static void run_phys()
+{
+  // TODO
+}
+
+static void run_bfb()
+{
+  static constexpr Int max_pack_size = 16;
+  REQUIRE(Spack::n <= max_pack_size);
+
+  // This is the threshold for whether the qc and qr cloud mixing ratios are
+  // large enough to affect the warm-phase process rates qcacc and ncacc.
+  constexpr Scalar qsmall = C::QSMALL;
+
+  constexpr Scalar t_freezing = 0.9 * C::RainFrze,
+                   t_not_freezing = 2.0 * C::RainFrze;
+  constexpr Scalar qr_incld_small = 0.9 * qsmall;
+  constexpr Scalar qr_incld_not_small = 2.0 * qsmall;
+  constexpr Scalar lamr1 = 0.1, lamr2 = 0.2, lamr3 = 0.3, lamr4 = 0.4;
+  constexpr Scalar mu_r1 = 0.2, mu_r2 = 0.4, mu_r3 = 0.6, mu_r4 = 0.8;
+  constexpr Scalar cdistr1 = 0.25, cdistr2 = 0.5, cdistr3 = 0.75, cdistr4 = 1.0;
+
+  RainImmersionFreezingData rain_imm_freezing_data[max_pack_size] = {
+    // t, lamr, mu_r, cdistr, qr_incld, qrheti, nrheti
+    {t_not_freezing, lamr1, mu_r1, cdistr1, qr_incld_small, 0.0, 0.0},
```

Comment:
If you leave off the outputs, they will automatically initialize to zero. It would be helpful to remove the last two arguments in all these initialization lists so that it's clear to all that you aren't using 0.0 as inputs (which would be really bad for unit testing). This also may need to be changed in your previous PRs.

Reply:
Okay. Should I submit a separate PR for fixing the other issues, or can they piggyback on this one?

Reply:
Piggy